code/Control/UserDefinedFormController.php

```diff
@@ -262,25 +265,42 @@ public function process($data, $form)
             if (!empty($data[$field->Name])) {
                 if (in_array(EditableFileField::class, $field->getClassAncestry() ?? [])) {
                     if (!empty($_FILES[$field->Name]['name'])) {
-                        $foldername = $field->getFormField()->getFolderName();
-
-                        // create the file from post data
-                        $upload = Upload::create();
-                        try {
-                            $upload->loadIntoFile($_FILES[$field->Name], null, $foldername);
-                        } catch (ValidationException $e) {
-                            $validationResult = $e->getResult();
-                            foreach ($validationResult->getMessages() as $message) {
-                                $form->sessionMessage($message['message'], ValidationResult::TYPE_ERROR);
+                        $doReturn = false;
+
+                        if (!$field->getFolderExists()) {
+                            $field->createProtectedFolder();
+                        }
+                        
+                        $file = Versioned::withVersionedMode(function () use ($field, $form, $doReturn) {
+                            $stage = Injector::inst()->get(self::class)->config()->get('file_upload_stage');
+                            Versioned::set_stage($stage);
+
+                            $foldername = $field->getFormField()->getFolderName();
+                            // create the file from post data
+                            $upload = Upload::create();
+                            try {
+                                $upload->loadIntoFile($_FILES[$field->Name], null, $foldername);
+                            } catch (ValidationException $e) {
+                                $validationResult = $e->getResult();
+                                foreach ($validationResult->getMessages() as $message) {
+                                    $form->sessionMessage($message['message'], ValidationResult::TYPE_ERROR);
+                                }
+                                Controller::curr()->redirectBack();
+                                $doReturn = true;
+                                return;
```

Comment:
```suggestion
                                return null;
```
Can get run of `$doReturn` entirely 


Once that's done change the `if ($doReturn) { return }` block below to  `if (is_null($file)) { return; }`